src/unused_finder/unused_finder.rs

```diff
@@ -164,122 +125,155 @@ impl UnusedFinder {
+            ));
+        }
+
+        // Get file paths only reachable by test files
+        let test_only_used_files: Vec<String> = graph
             .files
             .iter()
-            .filter_map(|(file_path, info)| {
-                if info.is_used {
-                    match file_path_exported_items_map.remove(file_path) {
-                        Some(mut exported_items) => {
-                            let unused_exports = &info.unused_exports;
-                            if unused_exports.is_empty() {
-                                return None;
-                            }
-                            let unused_exports = exported_items
-                                .drain(0..)
-                                .filter(|exported| {
-                                    unused_exports
-                                        .iter()
-                                        .any(|unused| unused.to_string() == exported.id.to_string())
-                                })
-                                .collect();
-                            return Some((file_path.to_string(), unused_exports));
-                        }
-                        None => return None,
+            .filter_map(|(file_path, graph_file)| {
+                // If after a graph bfs on test files is marked and used
+                // And file list of reachable files from entries contains said file
+                // said file is reachable only by test files
+                // If a file file was previously marked as unused, and graph_file.is_used == true after bfs on test files
+                if graph_file.is_used && !unused_prod_files.contains_key(file_path) {
+                    // If allow list contains a file or pattern that matches said file, we don't include it in the list
+                    if !allow_list.iter().any(|p| p.matches(&file_path)) {
+                        return Some(file_path.to_string());
                     }
                 }
                 None
             })
             .collect();
 
+        unused_items.iter_mut().for_each(|(file, items)| {
+            match graph.files.get(file) {
+                Some(graph_file) => {
+                    // Iterate over each unused item from entry files
+                    for item in items.iter_mut() {
+                        // If the item is no longer in the list of unused items of graph_file, it was used only by test files
+                        if !graph_file
+                            .unused_exports
+                            .iter()
+                            .any(|export| export.to_string() == item.id)
+                        {
+                            // Mark item as only used by tests
+                            item.test_only_use = true;
+                        }
+                    }
+                }
+                None => {}
+            }
+        });
+        // Retain only items that are not from files marked to be allowed-unused
```

Comment:
Should this list be transitive? e.g. if a file is allowed as unused, should its imports also be allowed?

Reply:
You mean, allow the imported files to also be unused?